src/Dropdown/index.jsx

```diff
@@ -1,225 +1,146 @@
+    if (prevState.open !== this.state.open) {
+      if (this.state.open) {
+        this.focusFirst();
+      } else {
+        this.toggleButton.focus();
+      }
     }
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    if (this.state.open) {
-      this.menuItems[this.state.focusIndex].focus();
-    } else if (prevState.open && this.toggleElem) {
-      this.toggleElem.focus();
-    }
+  componentWillUnmount = () => {
+    document.removeEventListener('click', this.handleDocumentClick, true);
   }
 
-  addEvents() {
-    document.addEventListener('click', this.handleDocumentClick, true);
+  getFocusableElements = () => {
+    const selector = 'button:not([disabled]), [href]:not([disabled]), input:not([disabled]), select:not([disabled]), textarea:not([disabled]), [tabindex]:not([tabindex="-1"]):not([disabled])';
+    return this.menuItems.current.querySelectorAll(selector);
   }
 
-  removeEvents() {
-    document.removeEventListener('click', this.handleDocumentClick, true);
+  focusFirst = () => {
+    this.getFocusableElements()[0].focus();
+  }
+
+  focusNext = () => {
+    const allFocusableElements = Array.from(this.getFocusableElements());
+    const activeIndex = allFocusableElements.indexOf(document.activeElement);
+    const nextIndex = (activeIndex + 1) % allFocusableElements.length;
+    allFocusableElements[nextIndex].focus();
+  }
+
+  focusPrevious = () => {
+    const allFocusableElements = Array.from(this.getFocusableElements());
+    const activeIndex = allFocusableElements.indexOf(document.activeElement);
+    const previousIndex =
+      ((activeIndex - 1) + allFocusableElements.length) % allFocusableElements.length;
+    allFocusableElements[previousIndex].focus();
   }
 
-  handleDocumentClick(e) {
+  handleDocumentClick = (e) => {
     if (this.container && this.container.contains(e.target) && this.container !== e.target) {
       return;
     }
-    this.toggle();
-  }
-
-  handleMenuKeyDown(e) {
-    e.preventDefault();
-    if (Dropdown.isTriggerKey('CLOSE_MENU', e.key)) {
+    if (this.state.open) {
       this.toggle();
-    } else if (Dropdown.isTriggerKey('SELECT_MENU_ITEM', e.key)) {
-      e.target.click();
-      this.setState({
-        open: false,
-      });
-    } else if (Dropdown.isTriggerKey('NAVIGATE_DOWN', e.key)) {
-      this.setState({
-        focusIndex: (this.state.focusIndex + 1) % this.props.menuItems.length,
-      });
-    } else if (Dropdown.isTriggerKey('NAVIGATE_UP', e.key)) {
-      this.setState({
-        focusIndex: ((this.state.focusIndex - 1) + this.props.menuItems.length) %
-                    this.props.menuItems.length,
-      });
     }
   }
 
-  handleToggleKeyDown(e) {
-    if (!this.state.open && Dropdown.isTriggerKey('OPEN_MENU', e.key)) {
-      this.toggle();
-    } else if (this.state.open && Dropdown.isTriggerKey('CLOSE_MENU', e.key)) {
-      this.toggle();
+  handleMenuKeyDown = (e) => {
+    switch (e.key) {
+      case 'ArrowUp':
```

Comment:
Not sure this will work for IE / Edge: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

Reply:
Looks like React has done this normalization for us: https://reactjs.org/docs/events.html#supported-events